src/obstacle_avoidance/src/MyNode.cpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * Created By: Gareth Ellis
+ * Created On: July 16th, 2016
+ * Description: An example node that subscribes to a topic publishing strings,
+ *              and re-publishes everything it receives to another topic with
+ *              a "!" at the end
+ */
+
+#include <MyNode.h>
+
+sensor_msgs::LaserScan AvoidObstacleClass::lidar_data;
+
+int sizeOfLidarArray(sensor_msgs::LaserScan lidar);
```

Comment:
`.cpp` files should almost always only have function **definitions**, not function **declarations**, so this should be moved to the `.h` file. Also, you might just not need this function, seeing as you can get the laserscan size with `lidar.ranges.size()`